Kernel/Syscalls/process.cpp

```diff
@@ -64,4 +64,23 @@ int Process::sys$set_process_name(Userspace<const char*> user_name, size_t user_
     return 0;
 }
 
+int Process::sys$set_coredump_metadata(Userspace<const Syscall::SC_set_coredump_metadata_params*> user_params)
+{
+    Syscall::SC_set_coredump_metadata_params params;
+    if (!copy_from_user(&params, user_params))
+        return -EFAULT;
+    if (params.key.length == 0)
+        return -EINVAL;
+    if (params.value.length == 0)
+        return -EINVAL;
+    auto copied_key = copy_string_from_user(params.key.characters, params.key.length);
+    if (copied_key.is_null())
+        return -EFAULT;
+    auto copied_value = copy_string_from_user(params.value.characters, params.value.length);
+    if (copied_value.is_null())
+        return -EFAULT;
+    m_coredump_metadata.set(move(copied_key), move(copied_value));
```

Comment:
Let's impose some limitations here so a userspace process can't fill the kernel heap with however much memory it wants.
Just some random numbers are fine, e.g maximum 16 keys, maximum 16KB per value (or whatever you think makes sense.)

Reply:
Seems sensible :+1:
16 entries / 16 KiB each should be more than enough.